api/v1alpha1/obclusteroperation_webhook.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"regexp"
+	"strings"
+
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/oceanbase/ob-operator/api/constants"
+	apitypes "github.com/oceanbase/ob-operator/api/types"
+	oceanbaseconst "github.com/oceanbase/ob-operator/internal/const/oceanbase"
+)
+
+// log is for logging in this package.
+var obclusteroperationlog = logf.Log.WithName("obclusteroperation-resource")
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (r *OBClusterOperation) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-oceanbase-oceanbase-com-v1alpha1-obclusteroperation,mutating=true,failurePolicy=fail,sideEffects=None,groups=oceanbase.oceanbase.com,resources=obclusteroperations,verbs=create;update,versions=v1alpha1,name=mobclusteroperation.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &OBClusterOperation{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *OBClusterOperation) Default() {
+	ctx := context.Background()
+	obcluster := OBCluster{}
+	err := clt.Get(ctx, types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      r.Spec.OBCluster,
+	}, &obcluster)
+	if err != nil {
+		obclusteroperationlog.Info("obcluster not found", "name", r.Spec.OBCluster)
+		return
+	}
+	if r.Labels == nil {
+		r.Labels = make(map[string]string)
+	}
+	r.Labels[oceanbaseconst.LabelRefOBCluster] = obcluster.Name
+	r.Spec.Type = apitypes.ClusterOperationType(strings.ToUpper(string(r.Spec.Type[0])) + string(r.Spec.Type[1:]))
```

Comment:
what's this for, ClusterOperationType already defines as beginning with capital letter

Reply:
It forgives types like `modifyOBServers`, which is the same as names of configuration items. I thought users will probably fill the `type` as a lower case.

Reply:
This is handy for users, why not leave it so?